miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultServiceRecordService.java

```diff
@@ -0,0 +1,113 @@
+import uk.ac.bbsrc.tgac.miso.core.store.ServiceRecordStore;
+import uk.ac.bbsrc.tgac.miso.service.InstrumentService;
+import uk.ac.bbsrc.tgac.miso.service.ServiceRecordService;
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizationManager;
+
+@Transactional(rollbackFor = Exception.class)
+@Service
+public class DefaultServiceRecordService implements ServiceRecordService {
+
+  private static final Logger log = LoggerFactory.getLogger(DefaultServiceRecordService.class);
+
+  @Autowired
+  private AuthorizationManager authorizationManager;
+
+  @Autowired
+  private MisoFilesManager misoFilesManager;
+
+  @Autowired
+  private ServiceRecordStore serviceRecordDao;
+
+  @Autowired
+  private DeletionStore deletionStore;
+
+  @Autowired
+  private InstrumentService instrumentService;
+
+  @Override
+  public Collection<ServiceRecord> listByInstrument(long instrumentId) throws IOException {
+    return serviceRecordDao.listByInstrumentId(instrumentId);
+  }
+
+  @Override
+  public ServiceRecord get(long recordId) throws IOException {
+    return serviceRecordDao.get(recordId);
+  }
+
+  @Override
+  public long create(ServiceRecord record) throws IOException {
+    record.setInstrument(instrumentService.get(record.getInstrument().getId()));
+    return serviceRecordDao.save(record);
+  }
+
+  @Override
+  public void update(ServiceRecord record) throws IOException {
+    ServiceRecord managed = get(record.getId());
+    applyRecordChanges(managed, record);
+    record.setInstrument(instrumentService.get(managed.getInstrument().getId()));
+    serviceRecordDao.save(managed);
+  }
+
+  private void applyRecordChanges(ServiceRecord target, ServiceRecord source) {
+    target.setTitle(source.getTitle());
+    target.setDetails(source.getDetails());
+    target.setServicedByName(source.getServicedByName());
+    target.setReferenceNumber(source.getReferenceNumber());
+    target.setServiceDate(source.getServiceDate());
+    target.setShutdownTime(source.getShutdownTime());
+    target.setRestoredTime(source.getRestoredTime());
+  }
+
+  @Override
+  public Map<String, Integer> getColumnSizes() throws IOException {
+    return serviceRecordDao.getServiceRecordColumnSizes();
+  }
+
+  public void setServiceRecordDao(ServiceRecordStore serviceRecordDao) {
+    this.serviceRecordDao = serviceRecordDao;
+  }
+
+  @Override
+  public DeletionStore getDeletionStore() {
+    return deletionStore;
+  }
+
+  @Override
+  public AuthorizationManager getAuthorizationManager() {
+    return authorizationManager;
+  }
+
+  @Override
+  public void afterDelete(ServiceRecord object) {
+    // Attempts to delete all attachments associated with a service record. Does not throw an exception upon failure to delete a file,
+    // but will log any failures
+    long recordId = object.getId();
+    try {
+      for (String filename : misoFilesManager.getFileNames(ServiceRecord.class, String.valueOf(recordId))) {
+        try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)